File: Scripts/micro_dao/contrato.py
from datetime import datetime

from micro_dao.objeto_modelo import ObjetoModelo


class Contrato(ObjetoModelo):
	_id: int
	_data_inicio: datetime
	_data_fim: datetime
	_data_ult_modif: datetime
	_descricao: str
	_horas_contratadas: int
	_fk_usuario: int

	def __init__(self, data_inicio: datetime, data_fim: datetime, data_ult_modif: datetime,
	             descricao: str, horas_contratadas: int, fk_usuario: int):
		self.set_data_inicio(data_inicio)
		self.set_data_fim(data_fim)
		self.set_data_ultima_modif(data_ult_modif)
		self.set_descricao(descricao)
		self.set_horas_contratadas(horas_contratadas)
		self.set_fk_usuario(fk_usuario)

	def get_id(self) -> int:
		return self._id

	def set_id(self, id: int):
		self._id = id
		return self

	def set_data_inicio(self, data_contrato: datetime):
		self._data_inicio = data_contrato

	def get_data_inicio(self) -> datetime:
		return self._data_inicio

	def set_data_fim(self, data_fim_prest: datetime):
		self._data_fim = data_fim_prest

	def get_data_fim(self) -> datetime:
		return self._data_fim

	def set_data_ultima_modif(self, data_ult_modif: datetime):
		self._data_ult_modif = data_ult_modif

	def get_data_ultima_modif(self) -> datetime:
		return self._data_ult_modif

	def get_descricao(self) -> str:
		return self._descricao

	def set_descricao(self, descricao: str):
		self._descricao = descricao
		return self

	def get_fk_usuario(self) -> int:
		return self._fk_usuario

	def set_fk_usuario(self, fk_usuario: int):
		self._fk_usuario = fk_usuario
		return self

	def set_horas_contratadas(self, horas_contratadas: int):
		self._horas_contratadas = horas_contratadas
		return self

	def get_horas_contratadas(self) -> int:
		return self._horas_contratadas

File: Projeto_Java/src/Dominio/Entidades/Contrato.java
package Dominio.Entidades;

import javax.persistence.*;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

@Entity
@Table(name = "contrato")
public class Contrato {

    @Id
    @Column(name = "id")
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int id;

    @Column(name = "data_inicio")
    private LocalDate dataInicio;

    @Column(name = "data_fim")
    private LocalDate dataFim;

    @Column(name = "data_ult_modif")
    private LocalDateTime dataUltimaModif;

    @Column(name = "descricao", length = 1000)
    private String descricao;

    @Column(name = "horas_contratadas")
    private int horasContratadas;

    @Column(name = "fk_usuario")
    private int fkUsuario;


    public Contrato(int fkUsuario) {
        descricao = "";
        dataInicio = null;
        dataFim = null;
        dataUltimaModif = null;
        horasContratadas = 0;
        this.fkUsuario = fkUsuario;
    }

    public Contrato(String descricao, LocalDate dataInicio, LocalDate dataFim,
                    int horasContratadas, int fkUsuario) {
        this.setDescricao(descricao);
        this.setDataInicio(dataInicio);
        this.setDataFim(dataFim);
        this.setHorasContratadas(horasContratadas);
        this.setFkUsuario(fkUsuario);
    }

    public Contrato(int id, String descricao, LocalDate dataInicio, LocalDate dataFim,
                    int horasContratadas, int fkUsuario) {
        this.setId(id);
        this.setDescricao(descricao);
        this.setDataInicio(dataInicio);
        this.setDataFim(dataFim);
        this.setHorasContratadas(horasContratadas);
        this.setFkUsuario(fkUsuario);
    }

    public Contrato(String descricao, LocalDate dataInicio, LocalDate dataFim,
                    LocalDateTime dataUltimaModif, int horasContratadas, int fkUsuario) {
        this.setDescricao(descricao);
        this.setDataUltimaModif(dataUltimaModif);
        this.setDataInicio(dataInicio);
        this.setDataFim(dataFim);
        this.setHorasContratadas(horasContratadas);
        this.setFkUsuario(fkUsuario);
    }

    public Contrato(int id, String descricao, LocalDate dataInicio, LocalDate dataFim,
                    LocalDateTime dataUltimaModif, int horasContratadas, int fkUsuario) {
        this.setId(id);
        this.setDescricao(descricao);
        this.setDataUltimaModif(dataUltimaModif);
        this.setDataInicio(dataInicio);
        this.setDataFim(dataFim);
        this.setHorasContratadas(horasContratadas);
        this.setFkUsuario(fkUsuario);
    }


    public int getId() {
        return id;
    }

    public void setId(int idContrato) {
        this.id = idContrato;
    }

    public String getDescricao() {
        return descricao;
    }

    public void setDescricao(String descricao) {

        if (!Util.isStringValida(descricao)) Util.throwExceptCampoVazio("descrição");

        this.descricao = descricao;
    }

    public LocalDateTime getDataUltimaModif() {
        return dataUltimaModif;
    }

    public void setDataUltimaModif(LocalDateTime dataUltimaModificacao) {
        this.dataUltimaModif = dataUltimaModificacao;
    }

    public LocalDate getDataInicio() {
        return dataInicio;
    }

    public void setDataInicio(LocalDate dataInicioPrestacao) {
        this.dataInicio = dataInicioPrestacao;
    }

    public LocalDate getDataFim() {
        return dataFim;
    }

    public void setDataFim(LocalDate dataFimPrestacao) {
        this.dataFim = dataFimPrestacao;
    }

    public int getFkUsuario() {
        return fkUsuario;
    }

    public void setFkUsuario(int fkUsuario) {
        this.fkUsuario = fkUsuario;
    }

    public int getHorasContratadas() {
        return horasContratadas;
    }

    public void setHorasContratadas(int horasContratadas) {
        this.horasContratadas = horasContratadas;
    }

    @Override
    public String toString() {
        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
        String contrato = "\nID:\t\t\t\t\t" + this.getId();
        contrato += "\nDescrição:\t\t\t" + this.getDescricao();
        contrato += "\nData Ult. Modif.:\t" + this.getDataUltimaModif().format(formatter);
        contrato += "\nData Início:\t\t" + this.getDataInicio();
        contrato += "\nData Fim:\t\t\t" + this.getDataFim();
        contrato += "\nHoras Contratadas:\t" + this.getHorasContratadas();
        contrato += "\nFK_usuario:\t\t\t" + this.getFkUsuario();
        return contrato;
    }
}
